release.md

```diff
@@ -0,0 +1,167 @@
+description and explanations of the context of preparing, creating and releasing
+new software versions.
+
+![general releasing context](releasing/images/general_context.png)
+
+### Releasing platforms supported
+
+The Ignition software supports a large variety of platforms where it can be
+built from source. A small subgroup of them is selected to be the officially
+supported for binaries distribution:
+
+ * **Linux:** .deb packages are being created for different Ubuntu and Debian
+   releases. Ubuntu LTS and Debian stable are officially supported.
+   Architectures supported are typically `amd64`, `arm64` and `armhf` but that
+   combination may not be available for all Ubuntu/Debian/Ignition releases.
+
+ * **MacOSX:** binaries for all active releases of MacOSX are usually created by
+   the Ignition development team under the support of
+   [Homebrew package manager](https://brew.sh/). Binary packages for Brew are
+   called `bottles`.
+
+ * **Windows:** currently *there is no binary distribution for Windows* that cover
+   all Ignition libraries although there is an ongoing effort for getting
+   binary packages inside
+   [conda-forge](https://github.com/conda-forge/staged-recipes/issues/13551)
+
+### Infrastructure
+
+The Ignition team uses different pieces of infrastructure to execute the
+releasing process:
+
+ * **build.osrfoundation.org**: main Jenkins server to orchestrate the whole
+   releasing process. It receives requests from Ignition developers to start
+   release process, schedule nightly builds, build the binary packages,
+   upload releasing artifacts to repositories, etc.
+
+ * **packages.osrfoundation.org**: Debian/Ubuntu .deb repositories.
+
+ * **osrf-distributions S3**: [cloud storage](https://osrf-distributions.s3.amazonaws.com/bottles-simulation)
+   space to save the different artifacts produced during the release process.
+
+### Type of releases
+
+The Ignition development uses three different types of releases depending on the
+stability of the software:
+
+ * **stable**: stable releases are designed to be generally consumed by users
+   that does not expect to solve severe issues or regressions.
+   * Linux stable binaries:
+     http://packages.osrfoundation.org/gazebo/{ubuntu/debian}-stable
+   * MacosX stable binaries:
+     https://osrf-distributions.s3.amazonaws.com/
+
+ * **prerelease**: prereleases are designed to be tested as candidates to
+   stable.
+   * Linux prerelease binaries:
+     http://packages.osrfoundation.org/gazebo/{ubuntu/debian}-prerelease
+   * MacosX prerelease binaries (same than stable):
+     https://osrf-distributions.s3.amazonaws.com/bottles-simulation
+
+ * **nightly**: automatically generated releases used in continuous integration
+   systems reflecting the latest changes in the repositories.
+   * Linux nightly binaries:
+     http://packages.osrfoundation.org/gazebo/{ubuntu/debian}-nightly
+   * MacosX nightly binaries:
+     *Unsupported*
```

Comment:
Maybe we could link to this issue: https://github.com/osrf/homebrew-simulation/issues/1314

Reply:
And I'm not sure if it's worth pointing out somewhere that we have unstable macOS releases with version `N-1.999.999~rev~date~commit` that aren't nightly generated, but are available

Reply:
551222b. I need to find a good place for special versioning of upstream snapshots. Probably in a new PR, we have more than enough here.